### PR TITLE
Added SOC paragraph

### PR DESCRIPTION
**Pull request by mulles**

### CODE REVIEW

**Review by mulles — COMMENTED** [review]
corrected some typos

File: docs/system/bms.md [hunk 1]
```diff
@@ -5,11 +5,20 @@ In most cases, the battery comes in a **pack**, which consists of multiple **mod
...
+
 ### Electric Cell Protection
 
 As described in chapter ["Battery"](battery.md), different cells have different minimum and maximum voltage levels as well as different phases while charging/discharging.
-The BMS is responsible to measure the voltage, current and temperature and calculate the state of charge (SOC) of the cells are in order to apply the correct current or voltage.
-It will optimise the charging rate and determine when to stop charging/discharging.
+
+The BMS is responsible to measure the voltage, current and temperature and stop or reduce dis-/charging in other to stay within defined safety limits of the cells.
```
Comment (martinjaeger) [inline docs/system/bms.md]:
```suggestion
The BMS is responsible to measure the voltage, current and temperature and stop or reduce dis-/charging in order to stay within defined safety limits of the cells.
```

File: docs/system/bms.md [hunk 2]
```diff
@@ -5,11 +5,20 @@ In most cases, the battery comes in a **pack**, which consists of multiple **mod
 
 ## Main Functions
 
+### State of Charge
+
+The BMS will optimize the charging rate and determine when to stop dis-/charging based on the state of charge (SOC).
+
+The SOC of the cells are calculated by combining current and voltage measurements, sometimes temperature. The most simple SOC determination is done with a open circuit voltage (OCV) lookup table. 
+It has a significant deviation due to the OCV not being accurately measurable during current flow. A more sophisticated algorithm consists of combining the voltage measurement with a current measurement, accumulating the current into or out of a cell, also know as coulomb counting.
```
Comment (martinjaeger) [inline docs/system/bms.md]:
```suggestion
The SOC of the cells is calculated by combining current and voltage measurements, sometimes temperature.

The most simple SOC determination is done with an open circuit voltage (OCV) lookup table. 
It has a significant deviation due to the OCV not being accurately measurable during current flow. A more sophisticated algorithm consists of combining the voltage measurement with a current measurement, accumulating the current into or out of a cell, also known as coulomb counting. 
```

File: docs/system/bms.md [hunk 3]
```diff
@@ -5,11 +5,20 @@ In most cases, the battery comes in a **pack**, which consists of multiple **mod
...
+
+The SOC of the cells are calculated by combining current and voltage measurements, sometimes temperature. The most simple SOC determination is done with a open circuit voltage (OCV) lookup table. 
+It has a significant deviation due to the OCV not being accurately measurable during current flow. A more sophisticated algorithm consists of combining the voltage measurement with a current measurement, accumulating the current into or out of a cell, also know as coulomb counting. 
+
+The current research on SOC estimation suggest, that a favourable algorithm to perform the combination of current and voltage is the extend kalman filter (EKF) or variation of it. This method is known as sensor fusion and is a well established. More advanced methods use an equivalent circuit model of the cell (ECM) in order to account for changing cell characteristics like internal resistance. These methods not only provide a better SOC, but also the state of health (SOH) of the battery. Good literature by Dr. Gregory L. Plett on implementation of such algorithms can be found [here](http://mocha-java.uccs.edu/ECE5720/index.html).
```
Comment (martinjaeger) [inline docs/system/bms.md]:
```suggestion
The current research on SOC estimation suggests that a favorable algorithm to perform the combination of current and voltage is the extended Kalman filter (EKF) or variations of it. This method is known as sensor fusion and is well established. More advanced methods use an equivalent circuit model of the cell (ECM) in order to account for changing cell characteristics like internal resistance. These methods not only provide a better SOC, but also the state of health (SOH) of the battery. Good literature by Dr. Gregory L. Plett on implementation of such algorithms can be found [here](http://mocha-java.uccs.edu/ECE5720/index.html).
```

**Review by martinjaeger — CHANGES_REQUESTED** [review]
Thanks, looks good. Just fixed some typos.

File: docs/system/bms.md (hunk 1 above)
Comment (mulles) [reply docs/system/bms.md]: Oops and thx

File: docs/system/bms.md (hunk 3 above)
Comment (mulles) [reply docs/system/bms.md]: thx for fixing.

**Comment by martinjaeger** [comment]
@mulles the issues are not yet resolved. You have to commit the suggestions. Just clicking "resolved" does not change anything. But I'll do that now, assuming you agree to the suggestions.

**Review by martinjaeger — APPROVED** [review]

**Pull request merged**